File: informatics_1/calc.cpp
#include <iostream>
#include <cctype>
#include <stack>

using namespace std; 

int main() {
    string s;    
    stack <int> st;  
    getline(cin, s);
    for (int i = 0; i < s.size(); i++) {
        if (s[i] == '+' || s[i] == '-' || s[i] == '*') {
        int a = st.top();
        st.pop();
        int b = st.top();
        st.pop();
        if (s[i] == '+')
            st.push(a + b);
        else if (s[i] == '-')
            st.push(b - a);
        else if (s[i] == '*')
            st.push(a * b);
        }
        else if (isdigit(s[i]))
            st.push(s[i] - '0');
    }
    cout << st.top() << '\n';
    return 0;
}

File: informatics_1/calc.py
stack = []
for element in input().split():
    if element in "+*-":
        a = stack[-2]
        b = stack[-1]
        # print(stack[-1])
        # print(stack[-2])
        stack.pop()
        stack.pop()
        if element == '+':
            result = a + b
        elif element == '-':
            result = a - b
        elif element == '*':
            result = a * b
        stack.append(result)
    else:
        stack.append(int(element))
print(stack[0])
